File: kalen/franc.py
class Franc:
    def __init__(self, amount):
        self.__amount = amount
    @property
    def amount(self):
        return self.__amount

    def times(self, multiplier: int):
        return Franc(self.amount * multiplier)
    
    def __eq__(self, object: object) -> bool:
        franc = object
        return self.amount == franc.amount

File: rinahira/php/src/Franc.php
<?php
declare(strict_types=1);
namespace app;

class Franc extends Money
{
    /**
     * 金額に数値を掛け金額を得る
     * 
     * @param integer $multiplier
     * @return Money
     */
    public function times(int $multiplier): Money
    {
        return Money::franc($this->amount * $multiplier);
    }
}
